arraylist.c

```diff
@@ -112,6 +112,33 @@ void arraylist_pop_and_copy(struct arraylist_t *arr, void *return_ptr)
     arraylist_rm(arr, arr->size - 1);
 }
 
+i32 arraylist_index_of(struct arraylist_t *arr, void *val, equality_fn_t *eq)
```

Comment:
Here we say the index will return an i32, but the arraylist can hold more items than that. Instead we should return a size_t here, and instead define NOT_FOUND to be SIZE_T_MAX or something

Reply:
Gucci